src/Microsoft.ML.Data/Prediction/Calibrator.cs

```diff
@@ -528,15 +528,15 @@ internal sealed class ParameterMixingCalibratedModelParameters<TSubModel, TCalib
         where TSubModel : class
         where TCalibrator : class, ICalibrator
     {
-        private readonly IPredictorWithFeatureWeights<float> _featureWeights;
+        private readonly TSubModel _featureWeights;
 
         internal ParameterMixingCalibratedModelParameters(IHostEnvironment env, TSubModel predictor, TCalibrator calibrator)
             : base(env, RegistrationName, predictor, calibrator)
         {
             Host.Check(predictor is IParameterMixer<float>, "Predictor does not implement " + nameof(IParameterMixer<float>));
             Host.Check(calibrator is IParameterMixer, "Calibrator does not implement " + nameof(IParameterMixer));
             Host.Assert(predictor is IPredictorWithFeatureWeights<float>);
```

Comment:
>Assert [](start = 17, length = 6)

Can this be a check instead of assert?

Reply:
On second thought, why does this class need to have a generic type?

---
In reply to: [310376340](https://github.com/dotnet/machinelearning/pull/4013#discussion_r310376340) [](ancestors = 310376340)

Reply:
This class needs to be generic to get the typed predictor and calibrator out of it. Otherwise OVA would just return this and it wouldn't be useful at all as far as getting the model goes. And I didn't do anything with the assert, so I am not sure why they did it that way in the beginning.

Reply:
This specific class is internal anyway, so if OVA uses it users will not be able to get the typed calibrator anyway. But I don't think OVA uses this class, does it?

---
In reply to: [310687802](https://github.com/dotnet/machinelearning/pull/4013#discussion_r310687802) [](ancestors = 310687802)

Reply:
So when OVA is returning the sub model parameters, if the model is calibrated it will return an instance of the `CalibratedModelParametersBase`. This class is public, and still allows access to the model parameters and the calibrator used for the training. Any changes made in this file were to facilitate this `CalibratedModelParametersBase` and returning the correct type, while still keeping the derived classes abstract.